src/core/datagram.c

```diff
@@ -31,6 +31,19 @@ QuicDatagramValidate(
     _In_ const QUIC_DATAGRAM* Datagram
     )
 {
+    QUIC_CONNECTION* Connection = QuicDatagramGetConnection(Datagram);
+    //
+    // If a datagram is to be sent down the connection, the datagram must have
+    // items in its queue. Otherwise, sending will have an error case.
+    //
+    if (!QuicConnIsClosed(Connection)) {
+        if ((Connection->Send.SendFlags & QUIC_CONN_SEND_FLAG_DATAGRAM) != 0) {
+            QUIC_DBG_ASSERT(Datagram->SendQueue != NULL);
+        } else if (Connection->State.PeerTransportParameterValid) {
+            QUIC_DBG_ASSERT(Datagram->SendQueue == NULL);
+        }
+    }
```

Comment:
If it is closed I think we should verify the flag is not set and the queue is empty too.

Reply:
Ok, done, and it looks like everything is building again now.